pkg/cmd/agent/nodeinfo_linux.go

```diff
@@ -17,9 +18,16 @@ func (h *nodeInfoHandler) sysInfo(info *scyllaclient.NodeInfo) error {
 		return err
 	}
 
+	var statfs unix.Statfs_t
+	if err := unix.Statfs(info.DataDirectory, &statfs); err != nil {
+		return errors.Wrap(err, "statfs")
+	}
+	total := statfs.Blocks * uint64(statfs.Bsize)
```

Comment:
Not an expert on this matter, but from my POV using cloud provider sdk (e.g. [ec2 sdk](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Client.DescribeVolumes)) should be safer and more convenient for the siren as well. Or perhaps do you think that it's completely unnecessary and would create the need to implement it separately for each provider in the `cloudmeta` pkg?

Reply:
Do you think NodeInfo is a good place to [extend](https://github.com/scylladb/scylla-manager/blob/164ff2ce065cd106253fed40d2f6ffe883e73cc6/v3/swagger/gen/agent/models/node_info.go#L18) with `InstanceType`, `CloudProvider` information? Or it's better to have a separate call for them?

Reply:
good idea actually, but I'm worried that these calls will require additional permissions (e.g. DescribeVolumes) that service_account (or iam role) that attached to instance should have. And in case of on-premise we need to query file system anyway

Reply:
We shouldn't worry with on-premise too much, as this effort is targeted for the cloud and other things won't work for on-premise as well (e.g. instance type). Having said that, I'm fine with the current approach, just wanted to note that there is another way of doing it, but if it doesn't seem better to you, there is no need for implementing it.